src/audio/module_adapter/library/userspace_proxy.c

```diff
@@ -27,14 +27,19 @@
 
 #include <sof/lib_manager.h>
 #include <sof/audio/component.h>
+#include <sof/schedule/dp_schedule.h>
 #include <rtos/userspace_helper.h>
 #include <utilities/array.h>
 #include <zephyr/sys/sem.h>
 #include <sof/audio/module_adapter/module/generic.h>
 
 #include <sof/audio/module_adapter/library/userspace_proxy.h>
+#include <sof/audio/module_adapter/library/userspace_proxy_user.h>
 #include <rimage/sof/user/manifest.h>
 
+ /* Assume that all the code runs in supervisor mode and don't made a system calls. */
+#define __ZEPHYR_SUPERVISOR__
```

Comment:
please enclose all these additions to the file in `#if CONFIG_SOF_USERSPACE_PROXY` or maybe move them out to a new file